index.js

```diff
@@ -1,28 +1,49 @@
 var app = require('express')();
+var bodyParser = require('body-parser');
 var request = require('request');
 var port = process.env.PORT || 3000;
 
-app.use(function(req, res, next) {
-  res.header('Access-Control-Allow-Origin', '*');
-  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
-  next();
-});
-
-app.get('/', function(req, res, next) {
+app.use(bodyParser.json());
 
-  if (req.query.url) {
-    request.get(req.query.url)
-    .on('error', console.log)
-    .on('response', function (response) {
+app.use(function (req, res, next) {
+	res.header('Access-Control-Allow-Origin', '*');
+	res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization');
+	next();
+});
 
-      // delete 'access-control-allow-origin' header to make sure response doesn't have CORS limits
-      delete response.headers['access-control-allow-origin'];
-    })
-    .pipe(res);
+app.options('*', function (req, res) {
+	res.send();
+});
 
-  } else {
-    res.send('please specify url query param');
-  }
+app.use(function (req, res, next) {
+	if (req.query.url) {
+		var options = {
+			url: req.query.url,
+			method: req.method,
+			headers: req.headers,
+			qs: req.query
+		};
+		if (['PUT', 'POST', 'PATCH'].indexOf(req.method) > -1) {
+			options.body = JSON.stringify(req.body);
```

Comment:
@mohsen1 I am guessing, setting Content-Type and Accept headers to 'application/json' would help in passing body in json form.